internal/services/v1/relationships_test.go

```diff
@@ -982,6 +984,86 @@ func TestDeleteRelationships(t *testing.T) {
+	}
+
+	iterations := 0
+	for i := 0; i < 10; i++ {
+		iterations++
+
+		headRev, err := ds.HeadRevision(context.Background())
+		require.NoError(err)
+
+		beforeDelete := readOfType(require, "document", client, zedtoken.MustNewFromRevision(headRev))
+
+		resp, err := client.DeleteRelationships(context.Background(), &v1.DeleteRelationshipsRequest{
+			RelationshipFilter: &v1.RelationshipFilter{
+				ResourceType: "document",
+			},
+			OptionalLimit:                 5,
+			OptionalAllowPartialDeletions: true,
+		})
+		require.NoError(err)
+
+		headRev, err = ds.HeadRevision(context.Background())
+		require.NoError(err)
+
+		afterDelete := readOfType(require, "document", client, zedtoken.MustNewFromRevision(headRev))
+		require.LessOrEqual(len(beforeDelete)-len(afterDelete), 5)
+
+		if i == 0 {
+			require.Equal(v1.DeleteRelationshipsResponse_DELETION_PROGRESS_PARTIAL, resp.DeletionProgress)
+		}
+
+		if resp.DeletionProgress == v1.DeleteRelationshipsResponse_DELETION_PROGRESS_COMPLETE {
+			require.NoError(err)
+			require.NotNil(resp.DeletedAt)
+
+			rev, err := zedtoken.DecodeRevision(resp.DeletedAt, ds)
+			require.NoError(err)
+			require.True(rev.GreaterThan(revision))
+			require.EqualValues(standardTuplesWithout(expected), readAll(require, client, resp.DeletedAt))
+			break
+		}
+	}
+
+	require.Less(iterations, 10)
```

Comment:
Why such a loose check? We have 11 rels, and are deleting them in batches of 5. That's 3 calls, so let's check we iterated 3 times

Reply:
Because I figured it was flexible if/when we change the test data (which happens from time to time). I can also make it strict, but this seemed simpler